ext/net/01_net.js

```diff
@@ -422,6 +422,14 @@ class Datagram {
       }
     }
   }
+
+  ref() {
+
+  }
+
+  unref() {
+
+  }
```

Comment:
Instead of exposing public APIs for ref and unref in `Deno.Datagram` it should be implemented internally. Something like this:

Add a new private fields  `this.#recvPromise = null` and `this.#unrefed = false` in `UdpWrap`.

In `#recieve`: https://github.com/denoland/deno/blob/3b397edf28d3ccfc42352a9f9d66059bc7bcb8be/ext/node/polyfills/internal_binding/udp_wrap.ts#L451

Change this line to store the current recieve promise in the field.

```js
this.#recvPromise = this.#listener!.receive(p);
if (this.#refed) core.unrefOpPromise(this.#recvPromise);

[buf, remoteAddr] = (await this.#recvPromise) as [
// ...
```

```js
// UdpWrap
ref() {
  if (this.#recvPromise) core.refOpPromise(this.#recvPromise);
  this.#unrefed = false;
}

unref() {
  if (this.#recvPromise) core.refOpPromise(this.#recvPromise);
  this.#unrefed = true;
}
```